Based on 6bc45b3b372bfc67a514d0bda24993e983f79aa8 Mon Sep 17 00:00:00 2001
From: Farook Al-Sammarraie <farook.a@scopicsoftware.com>
Date: Fri, 31 Oct 2025
Subject: [PATCH] Remove dependency on math.h. This of course should be fine,
 but there are some misconfigured cross compilation builds that have issues
 with math.h. These builds should be fixed, but I also just generally like to
 avoid dependencies where possible, which it is in this case.

--- a/include/fp16/fp16.h
+++ a/include/fp16/fp16.h
@@ -4,10 +4,8 @@
 
 #if defined(__cplusplus) && (__cplusplus >= 201103L)
 	#include <cstdint>
-	#include <cmath>
 #elif !defined(__OPENCL_VERSION__)
 	#include <stdint.h>
-	#include <math.h>
 #endif
 
 #ifdef _MSC_VER
@@ -228,9 +226,10 @@
 	const float scale_to_inf = fp32_from_bits(UINT32_C(0x77800000));
 	const float scale_to_zero = fp32_from_bits(UINT32_C(0x08800000));
 #endif
-	float base = (fabsf(f) * scale_to_inf) * scale_to_zero;
-
 	const uint32_t w = fp32_to_bits(f);
+	const float abs_f = fp32_from_bits(w & UINT32_C(0x7FFFFFFF));
+	float base = (abs_f * scale_to_inf) * scale_to_zero;
+
 	const uint32_t shl1_w = w + w;
 	const uint32_t sign = w & UINT32_C(0x80000000);
 	uint32_t bias = shl1_w & UINT32_C(0xFF000000);
